torch/csrc/Size.cpp

```diff
@@ -148,8 +148,25 @@ static PyObject *THPSize_numel(THPSize *self)
   END_HANDLE_TH_ERRORS
 }
 
+static PyObject *THPSize_reduce(THPSize *self)
+{
+  HANDLE_TH_ERRORS
+  auto ret = THPObjectPtr{PyTuple_New(2)};
```

Comment:
Almost every call here can fail and should be checked. For example, PyTuple_New can fail with due to out-of-memory. PyImport_ImportModule can fail for lots of reasons.

In general you need to write something like:

```
auto ret = THPObjectPtr{PyTuple_New(2)};
if (!ret) return NULL;
```

or you can write the following because of the HANDLE_TH_ERRORS macro:

```
auto ret = THPObjectPtr{PyTuple_New(2)};
if (!ret) throw python_error();
```

This is easy to mess up. I think it might be worth writing the `__reduce__` function in Python and binding it in C++, although that's not trivial either. (Up to you).